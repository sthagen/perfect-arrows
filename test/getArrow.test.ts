import { getArrow } from "../src"

describe("getArrow", () => {
  test("it computes an arrow between points", () => {
    expect(getArrow(10, 10, 30, 30)).toMatchSnapshot()
  })

  test("it computes an arrow from a point back to itself", () => {
    expect(getArrow(10, 10, 10, 10)).toMatchSnapshot()
  })

  test("it computes a bowed arrow between points", () => {
    expect(getArrow(10, 10, 30, 11, { bow: 1 })).toMatchSnapshot()
  })
})
