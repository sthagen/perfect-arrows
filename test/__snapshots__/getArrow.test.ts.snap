// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getArrow it computes a bowed arrow between points 1`] = `
Array [
  10,
  10,
  21.476103837238064,
  -19.022076744761318,
  30,
  11,
  1.2941671655185412,
  1.9473422795151378,
  0.049958395721942765,
]
`;

exports[`getArrow it computes a bowed arrow between points 1`] = `
Array [
  10,
  10,
  21.47616073286369,
  -19.02321465727379,
  30,
  11,
  1.2941671655185412,
  1.9473422795151378,
  0.049958395721942765,
]
`;

exports[`getArrow it computes an arrow between points 1`] = `
Array [
  10,
  10,
  20,
  20,
  30,
  30,
  0.7853981633974483,
  0.7853981633974483,
  0.7853981633974483,
]
`;

exports[`getArrow it computes an arrow from a point back to itself 1`] = `
Array [
  10,
  10,
  10,
  10,
  10,
  10,
  0,
  0,
  0,
]
`;
